### PR TITLE
Arc-1211 use new github client for github subscription delete

### PR DESCRIPTION
**Pull request by joshkay10**
Added the use of the user and app github clients to get installation and membership.

Since we have an authenticated token we can call getMembershipForAuthenticatedUser instead of getMembership which allows us to skip the step of a request to get user.

Added Nock to tests and increased test coverage by testing installation types Org and User, for both authorized and unauthorized paths.

Has not been tested in staging yet.(18/3/22 1300)

### CODE REVIEW

File: test/unit/frontend/__snapshots__/delete-github-subscription.test.ts.snap
Comment (joshkay10) [inline test/unit/frontend/__snapshots__/delete-github-subscription.test.ts.snap:1]: Auto generated file from the tests

File: test/unit/frontend/delete-github-subscription.test.ts
Comment (joshkay10) [inline test/unit/frontend/delete-github-subscription.test.ts:15]: Split Octokit out into its own test, since there was only 1 test using Octokit and it wasnt using Nock, was easier to have as a seperate describe block, have moved and updated existing tests inside the main describe.

File: src/routes/github/subscription/github-subscription-delete.ts
Comment (mboudreau) [inline src/routes/github/subscription/github-subscription-delete.ts:7]: once you merge with main, you should change these paths to the new shortened ones :)

File: src/routes/github/subscription/github-subscription-delete.ts
Comment (mboudreau) [inline src/routes/github/subscription/github-subscription-delete.ts:9]: Do you have a type for installation?

**Comment by mboudreau** [comment]
I'll approve after merge/cleanup :)

File: src/routes/github/subscription/github-subscription-delete.ts
Comment (joshkay10) [reply src/routes/github/subscription/github-subscription-delete.ts]: octokit Type added

**Review by mboudreau — DISMISSED** [review]

**Review by mboudreau — DISMISSED** [review]

**Review by mboudreau — DISMISSED** [review]

File: src/routes/github/subscription/github-subscription-delete.ts
Comment (mboudreau) [inline src/routes/github/subscription/github-subscription-delete.ts:41]: nice :)

File: src/services/github/user.ts
Comment (mboudreau) [inline src/services/github/user.ts:8]: Do we still need this the github client check?

**Review by mboudreau — APPROVED** [review]

File: src/services/github/user.ts
Comment (joshkay10) [reply src/services/github/user.ts:8]: For now, once the github client PR flag goes it can go too.

**Pull request merged**
